### ISSUE TITLE
Request for reopen #1727 Default params serialisation behaviour incompatible with recent Tomcat versions. 

### ISSUE THREAD

**Issue by tfactor2** [axios/axios#6503]
### Describe the bug

Request for reopening the [#1727 Default params serialisation behaviour incompatible with recent Tomcat versions.](https://github.com/axios/axios/issues/1727).

https://github.com/axios/axios/issues/1727#issuecomment-2244795901:

> It's unclear why it shouldn't be a default behavior - why not use standard encoding (RFC3986 / RFC1738 as qa does) as  
   expected without the need for doing workarounds?
   The problem is that one doesn't know in advance that Axios is working in such a way that might lead to unexpected bugs.

Original description
--------------------

> Summary
   Given the default params serialisation behaviour, axios out-of-the-box may fail to work with recent Tomcat versions out- 
   of-the-box.

> The problem is the way params are serialised in the absence of a custom params serialiser configuration. The code is 
   using the built-in encodeURIComponent function (good), but then proceeds to deliberately de-encode some characters. 
   In particular, it de-encodes [ and ] back to raw (problem).

> This is a problem with Tomcat because recent versions (e.g. 8.5.31+) have started being more pedantic about URI syntax, 
   and by default will reject requests that contain [ or ] in the query portion.

>  I appreciate the motive of making the URLs built by axios more human readable, and not percent-encoding characters if 
    you can get away with it. But blatant and wilful violation of RFC 3986, even if common practice for historical reasons, is a 
    problem with servers that are adopting strict-by-default behaviour.

 > Context
    axios version: v0.18.0
    Tomcat 8.5.32

reproduced on Axios 0.27.2.

### Axios Version

0.27.2

**Comment by xiic** [comment]
I agree with this, the current implementation to revert the encoding of some characters is indeed surprising, and also not documented. IIRC the characters `[` and `]` are considered unsafe and must be escaped.

Current logic (https://github.com/axios/axios/blob/v1.x/lib/helpers/buildURL.js):
```
return encodeURIComponent(val).
    replace(/%3A/gi, ':').
    replace(/%24/g, '$').
    replace(/%2C/gi, ',').
    replace(/%20/g, '+').
    replace(/%5B/gi, '[').
    replace(/%5D/gi, ']');
```

**Comment by jasonsaayman** [comment]
Fixed as a side effect of current parameter serialisation. Should this still be an issue, please open a new issue and tag me
